pytest_catchlog.py

```diff
@@ -162,59 +190,96 @@ def record_tuples(self):
 
             (logger_name, log_level, message)
         """
-        return [(r.name, r.levelno, r.getMessage()) for r in self.records()]
-
-    def set_level(self, level, logger=None):
-        """Sets the level for capturing of logs.
-
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
-        """
+        return [(r.name, r.levelno, r.getMessage()) for r in self.buffer]
 
-        obj = logger and logging.getLogger(logger) or self.handler
-        obj.setLevel(level)
+    @property
+    def buffer(self):
+        return self._handler.buffer
 
-    def at_level(self, level, logger=None):
-        """Context manager that sets the level for capturing of logs.
+    @property
+    def lines(self):
+        return [r.getMessage() for r in self.buffer]
 
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
+    def __init__(self, handler):
+        # Do not call LineMatcher.__init__(),
+        # it attempts to assign to self.lines, which is a property.
+        self._handler = handler
+
+    @classmethod
+    @contextmanager
+    def recording(cls, **kwargs):
+        with catching_logs(RecordingHandler(), **kwargs) as handler:
+            yield cls(handler)
+
+    def at_level(self, level):
+        """Context manager that sets the level for recording of logs.
+
+        This only affects logs collected by this funcarg. To control
+        captured logs that are reported for failed tests use the
+        ``logging_at_level()`` context manager.
```

Comment:
What's the rationale for dropping the `logger`/`handler` argument? Just a cleaner API?